### PR TITLE
Integration Test: Add error path testing for Auxiliary Image feature

### PR DESCRIPTION
**Pull request by xiancao**
Added the following negative tests case for auxiliary image feature. Make sure errors are reported clearly. 
(a) Missing common directory common mount path is not in the image
(b) missing domain config, what happens if the common mount contains only sparse JMS config
(c) common mount with out WDT binary
(d) Test a domain resource configuration using a custom common mount command that's guaranteed to fail, for example:  exit 1.

https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/5643/
https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/5640/
https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/5644/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonMiiTestUtils.java [hunk 1]
```diff
@@ -299,6 +299,52 @@ public static Domain createDomainResource(
...
+   * @param adminSecretName name of the new WebLogic admin credentials secret
+   * @param repoSecretName name of the secret for pulling the WebLogic image
+   * @param encryptionSecretName name of the secret used to encrypt the models
+   * @param replicaCount number of managed servers to start
+   * @param clusterName name of the cluster to add in domain
+   * @param auxiliaryImageVolumes list of AuxiliaryImageVolumes
+   * @param auxiliaryImages list of AuxiliaryImages
+   * @return domain object of the domain resource
+   */
+  public static Domain createDomainResource(
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonMiiTestUtils.java:318]: Can you rename the method name to createDomainResourceWithAuxiliaryImage()

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java [hunk 2]
```diff
@@ -307,6 +324,277 @@ public void testUpdateDataSourceInDomainUsingAuxiliaryImage() {
     logger.info("Found the DataResource configuration");
   }
 
+  /**
+   * Negative Test to create domain with mismatch mount path in auxiliary image and auxiliaryImageVolumes.
+   * in auxiliaryImageVolumes, set mountPath to "/errorpath"
+   * in auxiliary image, set AUXILIARY_IMAGE_PATH to "/auxiliary"
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:330]: Also mention that we check the event for associated warning message.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java [hunk 3]
```diff
@@ -307,6 +324,277 @@ public void testUpdateDataSourceInDomainUsingAuxiliaryImage() {
...
+
+    // create domain and verify it is failed
+    logger.info("Creating domain {0} with auxiliary image {1} in namespace {2}",
+        domainUid, errorPathAuxiliaryImage1, errorpathDomainNamespace);
+    assertDoesNotThrow(() -> createDomainCustomResource(domainCR), "createDomainCustomResource throws Exception");
+
+    // check the domain event contains the expected error msg
+    String expectedErrorMsg = "Failed to complete processing domain resource domain1 due to: "
+        + "Auxiliary Image: Dir '/errorpath' doesn't exist or is empty. Exiting.";
+    verifyDomainEventContainsExpectedErrorMsg(timestamp, expectedErrorMsg);
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:388]: Any error on operator/introspection  logs ?

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java [hunk 4]
```diff
@@ -307,6 +324,277 @@ public void testUpdateDataSourceInDomainUsingAuxiliaryImage() {
...
+        + "a WebLogic Deploy Tool (WDT) install is not located at  'spec.configuration.model.wdtInstallHome'  "
+        + "which is currently set to '/auxiliary/weblogic-deploy'";
+    verifyDomainEventContainsExpectedErrorMsg(timestamp, expectedErrorMsg);
+
+    // delete domain1
+    deleteDomainResource(errorpathDomainNamespace, domainUid);
+  }
+
+  /**
+   * Negative Test to create domain without domain model file, the auxiliary image contains only sparse JMS config.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:458]: Add the assertion we do here to check the warning.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java [hunk 5]
```diff
@@ -307,6 +324,277 @@ public void testUpdateDataSourceInDomainUsingAuxiliaryImage() {
...
+    String expectedErrorMsg = "Failed to complete processing domain resource domain1 due to: "
+        + "Auxiliary Image: Dir '/errorpath' doesn't exist or is empty. Exiting.";
+    verifyDomainEventContainsExpectedErrorMsg(timestamp, expectedErrorMsg);
+
+    // delete domain1
+    deleteDomainResource(errorpathDomainNamespace, domainUid);
+  }
+
+  /**
+   * Negative Test to create domain without WDT binary.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:395]: Add the assertion we do here to check the warning.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java [hunk 6]
```diff
@@ -307,6 +324,277 @@ public void testUpdateDataSourceInDomainUsingAuxiliaryImage() {
...
+    String expectedErrorMsg =
+        "createDomain did not find the required domainInfo section in the model file /auxiliary/models/model.jms2.yaml";
+    verifyDomainEventContainsExpectedErrorMsg(timestamp, expectedErrorMsg);
+
+    // delete domain1
+    deleteDomainResource(errorpathDomainNamespace, domainUid);
+  }
+
+  /**
+   * Negative Test to create domain using a custom mount command that's guaranteed to fail, for example:  exit 1.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:523]: Add the assertion we do here to check the warning.

Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:523]: Can u describe the usecase, where we put the commabd "exit 1"

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java [hunk 7]
```diff
@@ -400,4 +688,31 @@ private void createAuxiliaryImage(String stageDirPath, String dockerFileLocation
...
+    }
+
+    // create encryption secret
+    logger.info("Create encryption secret");
+    if (!secretExists(encryptionSecretName, domainNamespace)) {
+      createSecretWithUsernamePassword(encryptionSecretName, domainNamespace, "weblogicenc", "weblogicenc");
+    }
+  }
+
+  private void verifyDomainEventContainsExpectedErrorMsg(OffsetDateTime timestamp, String expectedErrorMsg) {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:711]: A candidate to be included in a generic utility class.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java [hunk 8]
```diff
@@ -400,4 +688,31 @@ private void createAuxiliaryImage(String stageDirPath, String dockerFileLocation
             .command(cmdToExecute))
         .execute(), String.format("Failed to execute", cmdToExecute));
   }
+
+  private void createSecretsForDomain(String adminSecretName, String encryptionSecretName, String domainNamespace) {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:692]: Can we not directly call the method createOcirRepoSecret() and createSecretWithUsernamePassword() directly with out creating a wrapper private method  createSecretsForDomain()

Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:692]: We have multiple test methods using the same piece of code. That's why I added a wrapper method.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java (hunk 6 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:523]: This is the usecase for (g) Test a domain resource configuration using a custom common mount command that's guaranteed to fail, for example:  exit 1.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonMiiTestUtils.java (hunk 1 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonMiiTestUtils.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java (hunk 2 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:332]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java (hunk 3 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java]: added checking introspector logs and operator logs.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java (hunk 4 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:479]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java (hunk 5 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:406]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java (hunk 7 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java]: moved to CommonTestUtils.java

**Review by vanajamukkara — CHANGES_REQUESTED** [review]
Can you add a check that domain is not restarted for all the negative cases?

**Review by vanajamukkara — APPROVED** [review]
LGTM

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java [hunk 9]
```diff
@@ -307,6 +329,350 @@ public void testUpdateDataSourceInDomainUsingAuxiliaryImage() {
...
+        "Warning", timestamp, expectedErrorMsg);
+
+    // check the operator pod log contains the expected error msg
+    String operatorPodName =
+        assertDoesNotThrow(() -> getOperatorPodName(OPERATOR_RELEASE_NAME, opNamespace));
+    checkPodLogContainsString(opNamespace, operatorPodName, expectedErrorMsg);
+
+    // verify the domain is not rolled
+    // TODO: enable this check once https://jira.oraclecorp.com/jira/browse/OWLS-90971 is fixed
+    /*
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:647]: Use the utility oracle.weblogic.kubernetes.utils.CommonMiiTestUtils.verifyPodsNotRolled;

Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:647]: yes, in the oracle.weblogic.kubernetes.utils.CommonMiiTestUtils.verifyPodsNotRolled, it only sleeps for 2 minutes. It is not enough in this case.

**Review by anpanigr — APPROVED** [review]
lgtm

**Pull request merged**